src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCNV.java

```diff
@@ -258,11 +254,99 @@ protected void runPipeline(final JavaSparkContext ctx) {
         final File finalModeledSegmentsFileAsGatkCNV = new File(outputPrefix + "-" + FINAL_SEG_FILE_TAG + "." + GATK_SEG_FILE_TAG + ".seg");
         modeller.writeModeledSegmentFile(finalModeledSegmentsFileAsGatkCNV);
 
-        // Write file for ACS- compatible output to help Broad CGA
+        //write file for ACS-compatible output to help Broad CGA
         final File finalACSModeledSegmentsFile = new File(outputPrefix + "-" + FINAL_SEG_FILE_TAG + "." + CGA_ACS_SEG_FILE_TAG + ".seg");
         modeller.writeAllelicCapSegFile(finalACSModeledSegmentsFile);
 
         ctx.setLogLevel(originalLogLevel);
         logger.info("SUCCESS: Allelic CNV run complete for sample " + sampleName + ".");
     }
-}
+
+    //merge copy-neutral and uncalled segments (unless disabled) and fix up target-segment start breakpoints
+    private List<SimpleInterval> prepareTargetSegments(final Genome genome,
+                                                       final List<ModeledSegment> targetSegmentsWithCalls) {
+        final List<SimpleInterval> targetSegmentsUnfixed;
+        if (!useAllCopyRatioSegments) {
+            logger.info("Merging copy-neutral and uncalled target-coverage segments...");
+            targetSegmentsUnfixed = SegmentMergeUtils.mergeNeutralSegments(targetSegmentsWithCalls);
+            logger.info("Number of segments after copy-neutral merging: " + targetSegmentsUnfixed.size());
+        } else {
+            targetSegmentsUnfixed = targetSegmentsWithCalls.stream().map(ModeledSegment::getSimpleInterval)
+                    .collect(Collectors.toList());
+        }
+        //fix up target-segment start breakpoints (convert from target-end--target-end to target-start--target-end)
+        return SegmentUtils.fixTargetSegmentStarts(targetSegmentsUnfixed, genome.getTargets());
+    }
+
+    //segment SNPs using CBS on per-SNP MLE minor allele fraction iteratively until convergence
+    //(final segment file is output as a side effect)
+    private List<SimpleInterval> performSNPSegmentationStep(final String sampleName, final Genome genome) {
+        //initial segmentation of SNPs on per-SNP MLE minor allele fraction, assuming no allelic bias
+        //(segments are written to temporary file)
+        logger.info("Performing initial SNP segmentation (assuming no allelic bias)...");
+        final File snpSegmentFile;
+        try {
+            snpSegmentFile = File.createTempFile(outputPrefix + "-" + SNP_MAF_SEG_FILE_TAG + "-tmp", ".seg");
+
+        } catch (final IOException e) {
+            throw new UserException.CouldNotCreateOutputFile("Could not create temporary SNP segmentation file.", e);
+        }
+        List<SimpleInterval> snpSegments = calculateAndWriteSNPSegmentation(sampleName, genome, snpSegmentFile, 1.);    //allelic bias = 1.
```

Comment:
[this is a hard call for me, ignore if you're not sold] In this context `1.` might actually be a magic constant that you should extract as `INITIAL_ALLELIC_BIAS_GUESS = 1.`  After all, one could reasonably start with something like `1.05`.